From 700c302adfeb1254596dbac5fbd589d9f0909bd2 Mon Sep 17 00:00:00 2001
From: Artur Wieczorek <artwik@wp.pl>
Date: Sun, 5 Apr 2015 11:38:51 +0200
Subject: Warn if wxUSE_RICHTEXT is set to 1 and in the same time wxUSE_HTML
 and/or wxUSE_LONGLONG is set to 0.

wxRichTextCtrl requires wxHTML and wxLongLong and if they are is missing then appropriate actions (warning or correction) are executed in chkconf.h.

diff --git a/include/wx/chkconf.h b/include/wx/chkconf.h
index 963d156..7451402 100644
--- a/include/wx/chkconf.h
+++ b/include/wx/chkconf.h
@@ -2269,6 +2269,26 @@
 #   endif
 #endif /* wxUSE_STC */
 
+#if wxUSE_RICHTEXT
+#   if !wxUSE_HTML
+#       ifdef wxABORT_ON_CONFIG_ERROR
+#           error "wxRichTextCtrl requires wxUSE_HTML"
+#       else
+#           undef wxUSE_RICHTEXT
+#           define wxUSE_RICHTEXT 0
+#       endif
+#   else
+#     if !wxUSE_LONGLONG
+#         ifdef wxABORT_ON_CONFIG_ERROR
+#             error "wxRichTextCtrl requires wxUSE_LONLONG"
+#         else
+#             undef wxUSE_LONLONG
+#             define wxUSE_LONLONG 1
+#         endif
+#     endif
+#   endif
+#endif /* wxUSE_RICHTEXT */
+
 #endif /* wxUSE_GUI */
 
 #endif /* _WX_CHKCONF_H_ */
-- 
1.9.5.msysgit.0

